### PR TITLE
 Migrate 'components/integrations/' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by A01123748**
#### Summary
This PR includes updating the files under components/integration to a typescript format and associated tests.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-20454

#### Related Pull Requests
This is a replacement for the original [PR#9271](https://github.com/mattermost/mattermost-webapp/pull/9271) which was closed by mistake.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by A01123748** [comment]
@hmhealey I hope this is what we need to get this resolved.

**Comment by hmhealey** [comment]
Yeah, it looks like it should be good to review again. I'll take a look at what's causing the tests to fail since I know they were working before I did my rebase, so it's likely my fault that they're broken again. I'll take a look first thing tomorrow morning.

Thanks again for putting up with me kind of destroying your branch there. It's pretty easy to do some stupid stuff with git, but it's definitely been a while since I've done something that dumb 😅

**Review by hmhealey — APPROVED** [review]
After all that, this looks good to me now!

File: components/integrations/abstract_incoming_hook.test.tsx [hunk 1]
```diff
@@ -4,25 +4,38 @@
 import React from 'react';
 import {shallow} from 'enzyme';
 
-import AbstractIncomingWebhook from 'components/integrations/abstract_incoming_webhook.jsx';
+import AbstractIncomingWebhook from 'components/integrations/abstract_incoming_webhook';
+import {Team} from 'packages/mattermost-redux/src/types/teams';
+import {IncomingWebhook} from 'mattermost-redux/types/integrations';
+import {TestHelper} from '../../utils/test_helper';
```
Comment (mickmister) [inline components/integrations/abstract_incoming_hook.test.tsx:10]: Same suggestion to reorder imports. I suppose this applies to many test files in this PR

```suggestion
import {Team} from 'mattermost-redux/src/types/teams';
import {IncomingWebhook} from 'mattermost-redux/types/integrations';
import {TestHelper} from 'utils/test_helper';

import AbstractIncomingWebhook from './abstract_incoming_webhook';
```

File: components/integrations/installed_outgoing_webhook.tsx [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
 
+import {OutgoingWebhook} from 'mattermost-redux/types/integrations';
+import {Team} from 'mattermost-redux/types/teams';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
+
 import DeleteIntegrationLink from './delete_integration_link';
 
-export function matchesFilter(outgoingWebhook, channel, filter) {
+export function matchesFilter(outgoingWebhook: any, channel: any, filter: string) {
```
Comment (mickmister) [inline components/integrations/installed_outgoing_webhook.tsx:17]:
```suggestion
export function matchesFilter(outgoingWebhook: OutgoingWebhook, channel: Channel, filter?: string) {
```

File: components/integrations/installed_incoming_webhook.tsx [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
 
+import {IncomingWebhook} from 'mattermost-redux/types/integrations';
+
 import {getSiteURL} from 'utils/url';
 
 import CopyText from 'components/copy_text';
 
+import {Team} from 'mattermost-redux/types/teams';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
```
Comment (mickmister) [inline components/integrations/installed_incoming_webhook.tsx:16]: Can we put these imports with the other `mattermost-redux/types` import above?

**Review by mickmister — APPROVED** [review]
Awesome work @A01123748, thanks for the contribution :tada:

LGTM. I just have some non-blocking suggestions on adjusting import orders

**Comment by jgilliam17** [comment]
Thanks @A01123748 
Running E2Es at the moment. I will post the link to report and review when run completes.

**Comment by jgilliam17** [comment]
Thank you @A01123748 Here are the results https://automation-dashboard.vercel.app/cycles/6ee895fa-b674-4a50-ab22-8edeaf252b99
Significant number of E2Es broke due to this change
When manually testing, there are errors when trying to add slash commands and webhooks, can you please take a look? Thanks
Screen Shot 2021-11-26 at 8 38 42 AM
Screen Shot 2021-11-26 at 8 36 33 AM

**Comment by hmhealey** [comment]
Hi @A01123748. Is there anything we can help with on this?

**Comment by A01123748** [comment]
Hi @hmhealey, just been pretty busy these days so I haven't been able to dedicate anytime to this. Hope to get some spare time soon.

**Comment by M-ZubairAhmed** [comment]
@A01123748 would you need some help to get this to finish line ? :)

**Comment by M-ZubairAhmed** [comment]
These have been split into multiple issues and prs

**Pull request closed without merging**